builds/Resources/parse_pre_kanata.py

```diff
@@ -0,0 +1,60 @@
+#! /usr/bin/env python3
+#
+#  Copyright (c) 2024 Franz Fuchs
+#  All rights reserved.
+# 
+#  This software was developed by the University of  Cambridge
+#  Department of Computer Science and Technology under the
+#  SIPP (Secure IoT Processor Platform with Remote Attestation)
+#  project funded by EPSRC: EP/S030868/1
+# 
+#  @BERI_LICENSE_HEADER_START@
+# 
+#  Licensed to BERI Open Systems C.I.C. (BERI) under one or more contributor
+#  license agreements.  See the NOTICE file distributed with this work for
+#  additional information regarding copyright ownership.  BERI licenses this
+#  file to you under the BERI Hardware-Software License, Version 1.0 (the
+#  "License"); you may not use this file except in compliance with the
+#  License.  You may obtain a copy of the License at:
+# 
+#    http://www.beri-open-systems.org/legal/license-1-0.txt
+# 
+#  Unless required by applicable law or agreed to in writing, Work distributed
+#  under the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR
+#  CONDITIONS OF ANY KIND, either express or implied.  See the License for the
+#  specific language governing permissions and limitations under the License.
+# 
+#  @BERI_LICENSE_HEADER_END@
+#
+
+import argparse
+
+def parse(filename):
+    with open(filename, "r") as f:
+        cur_cycle = 0
+        print("Kanata\t0004")
+        print("C=\t0")
+        for line in f:
+            v = line.split()
+            c = int(v[1])
```

Comment:
If I'm understanding this correctly, it seems a little fragile, e.g. if some other print fires then this could get konata out of sync or cause python to crash? It might be good to filter so that only lines containing some magic konata keyword get considered

Reply:
As we discussed, sounds like there's a mix of preprocessing that is expected to happen before the script gets called. Let's work out where the appropriate makefile etc should live, but the support can get merged first.